File: 07_김연우/2231.c
 
#include <stdio.h>
#include <math.h>


int GetDigit(int Num) {
	int Digit = 1;
	while (Num > 9) {
		Num /= 10;
		Digit++;
	}
	return Digit;
}


int GetSumDecomposition(int Num) {
	int Num_Sum = 0;
	int Num_Cnt = 0;
	int Num_Digit = GetDigit(Num);
	for (int i = 0; i < Num_Digit; i++) {
		Num_Sum += (Num % (int)(pow(10, Num_Digit - i)) - Num % (int)(pow(10, Num_Digit - (i + 1)))) / pow(10, Num_Digit - i - 1);
	}
	Num_Sum += Num;
	return Num_Sum;
}

int GetMinConstructor(int Num) {return Num - GetDigit(Num) * 9;}
int GetMaxConstructor(int Num) {return Num;}


int main(void)
{
	unsigned int Input, Input_Digit = 1, Input_Temp, Input_Sum = 0;
	unsigned int Output, Output_Max, Output_Min, Output_Cnt;
	scanf_s("%d", &Input);
	int Constructor_Min = GetMinConstructor(Input);
	int Constructor_Max = GetMaxConstructor(Input);
	for (Constructor_Min; Constructor_Min < Constructor_Max; Constructor_Min++) {
		if (GetSumDecomposition(Constructor_Min) == Input) { break; }
	}
	printf("%d\n", Constructor_Min);

}

File: 16_이찬희/2231.py
#분해합

n = int(input())

for i in range(1, n+1):  # map(function, iterable) : 함수, 반복가능한 자료형(리스트 , 튜플 등) / 반환값 : map 객체
    num = sum(map(int, str(i)))  #num에는 각 자리 수들의 합이 저장된다. 
    num_sum = i + num  # num_sum = 각 자리 수 합 + 숫자 = 분해합 

    if num_sum == n: 
        print(i)
        break
    if i == n: 
        print(0)
